src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -421,34 +432,16 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::_OnNetworkScanFinished
 template <class ImplClass>
 ConnectivityManager::ThreadDeviceType GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetThreadDeviceType(void)
 {
-    ConnectivityManager::ThreadDeviceType deviceType;
-
-    Impl()->LockThreadStack();
-
-    const otLinkModeConfig linkMode = otThreadGetLinkMode(mOTInst);
-
-#if CHIP_DEVICE_CONFIG_THREAD_FTD
-    if (linkMode.mDeviceType && otThreadIsRouterEligible(mOTInst))
-        ExitNow(deviceType = ConnectivityManager::kThreadDeviceType_Router);
-    if (linkMode.mDeviceType)
-        ExitNow(deviceType = ConnectivityManager::kThreadDeviceType_FullEndDevice);
-#endif
-    if (linkMode.mRxOnWhenIdle)
-        ExitNow(deviceType = ConnectivityManager::kThreadDeviceType_MinimalEndDevice);
-
-    ExitNow(deviceType = ConnectivityManager::kThreadDeviceType_SleepyEndDevice);
-
-exit:
-    Impl()->UnlockThreadStack();
-
-    return deviceType;
+    return mDeviceType;
```

Comment:
I do not believe this is correct. You are deleting the mean to differentiate between full thread device and router

Reply:
I left the initial version and only modified the MTD / Sleepy conditional.